content/blog/2021/Bitcoin_Core_RPC_Demo.md

```diff
@@ -0,0 +1,666 @@
+
+    // Generate a fresh mnemonic, and from there a privatekey
+    let mnemonic: GeneratedKey<_, Segwitv0> =
+                Mnemonic::generate((MnemonicType::Words12, Language::English)).unwrap();
+    let mnemonic = mnemonic.into_key();
+    let xkey: ExtendedKey = (mnemonic, password).into_extended_key().unwrap();
+    let xprv = xkey.into_xprv(Network::Regtest).unwrap();
+
+    // Create derived privkey from the above master privkey
+    // We use the following derivation paths for recieve and change keys
+    // receive: "m/84h/1h/0h/0"
+    // change: "m/84h/1h/0h/1" 
+    let mut keys = Vec::new();
+
+    for path in ["m/84h/1h/0h/0", "m/84h/1h/0h/1"] {
+        let deriv_path: DerivationPath = DerivationPath::from_str(path).unwrap();
+        let derived_xprv = &xprv.derive_priv(&secp, &deriv_path).unwrap();
+        let origin: KeySource = (xprv.fingerprint(&secp), deriv_path);
+        let derived_xprv_desc_key: DescriptorKey<Segwitv0> =
+        derived_xprv.into_descriptor_key(Some(origin), DerivationPath::default()).unwrap();
+
+        // Wrap the derived key with the wpkh() string to produce a descriptor string
+        if let Secret(key, _, _) = derived_xprv_desc_key {
+            let mut desc = "wpkh(".to_string();
+            desc.push_str(&key.to_string());
+            desc.push_str(")");
+            keys.push(desc);
+        }
+    }
+    
+    // Return the keys as a tupple
+    (keys[0].clone(), keys[1].clone())
+}
+```
+
+To check that the above added function is working as expected, call it in the main function and print the descriptors
+``` rust
+use ...
+
+fn main() {
+    let (receive_desc, change_desc) = get_descriptors();
+    println!("recv: {:#?}, \nchng: {:#?}", receive_desc, change_desc);
+}
+
+fn get_descriptors() -> (String, String) {
+    ...
+}
+```
+Running the binary should produces the following result
+```shell
+$ cargo run
+recv: "wpkh([89df6a67/84'/1'/0'/0]tprv8iSRXyLtTKJN9qt1jyPVqwhDMEaYztXunPaRQznaH1z8gj8e2g7RnF2ZoHP56VEXwMn76AiV1Je6nJmZbFistwAQCrRGmSrsoKfdqfTDNA1/*)", 
+chng: "wpkh([89df6a67/84'/1'/0'/1]tprv8iSRXyLtTKJNCECQxBJ19cgx2ueS7mC7GNq7VqTWY3RNPMBY7DfTb9HUnXpJqa14jCJNRmi4yGxfoTVS4WLBXDkvTLq4vujeAD9NfDtSxGP/*)"
+```
+Voila! Now we have nice descriptors strings handy to use for our BDK wallet construction.
+
+### Talking to Bitcoin Core Programmatically
+Like all other tutorials we will use two wallets to send coins back and forth. A Bitcoin Core wallet accessible via `bitcoin-cli` command line tools, and a BDK wallet maintained by BDK library.    
+
+But unlike other tutorial we won't be using `bitcoin-cli` to talk to the Core wallet (we can, but lets spice things up). Instead we will use the `bitcoin-rpc` library, to talk with our core node listening at `127.0.0.1:18443`, from inside our main function. This will allow us to write code, that will handle both the core and BDK wallet, from inside of the same function, and we wont have to switch terminals!
```

Comment:
```suggestion
But unlike other tutorials, we won't be using `bitcoin-cli` to talk to the Core wallet (we can, but let's spice things up). Instead, we will use the `bitcoin-rpc` library, to talk with our core node listening at `127.0.0.1:18443`, from inside our main function. This will allow us to write code, that will handle both the core and BDK wallet, from inside of the same function, and we won't have to switch terminals!
```